Cargo.toml

```diff
@@ -6,7 +6,13 @@ edition = "2021"
 [build-dependencies]
 lalrpop = "0.20.2"
 
+
 [dependencies]
+logos = "0.12.1"
+enum-as-inner = "0.6.0"
+serde = { version = "1.0.130", features = ["derive"] }
+serde_json = "1.0.61"
+strum_macros = "0.25.3"
```

Comment:
There seems to be several files with merge conflicts, I guess the refactoring work is changing much from when you started doing your work. It would be good to resolve.